test/framework/src/main/java/org/elasticsearch/transport/nio/NioClient.java

```diff
@@ -0,0 +1,164 @@
+    private final ChildSelectorStrategy strategy;
+    private final TimeValue defaultConnectTimeout;
+    private final ChannelFactory channelFactory;
+    private final AtomicInteger current = new AtomicInteger(0);
+
+    public NioClient(Logger logger, OpenChannels openChannels, ChildSelectorStrategy strategy, TimeValue connectTimeout,
+                     ChannelFactory channelFactory) {
+        this.logger = logger;
+        this.openChannels = openChannels;
+        this.strategy = strategy;
+        this.defaultConnectTimeout = connectTimeout;
+        this.channelFactory = channelFactory;
+    }
+
+    public boolean connectToChannels(DiscoveryNode node, NioSocketChannel[] channels, TimeValue connectTimeout,
+                                     Consumer<NioChannel> closeListener) throws IOException {
+        boolean allowedToConnect = acquireAccess();
+        if (allowedToConnect == false) {
+            return false;
+        }
+
+        try {
+            connectTimeout = getConnectTimeout(connectTimeout);
+            final ArrayList<NioSocketChannel> connections = new ArrayList<>(channels.length);
+            final InetSocketAddress address = node.getAddress().address();
+            for (int i = 0; i < channels.length; i++) {
+                SocketSelector socketSelector = strategy.next();
+                NioSocketChannel nioSocketChannel = channelFactory.openNioChannel(address);
+                openChannels.clientChannelOpened(nioSocketChannel);
+                nioSocketChannel.getCloseFuture().setListener(closeListener);
+                connections.add(nioSocketChannel);
+                socketSelector.registerSocketChannel(nioSocketChannel);
+            }
+
+            try {
+                Exception ex = null;
+                boolean allConnected = true;
+                for (NioSocketChannel socketChannel : connections) {
+                    ConnectFuture connectFuture = socketChannel.getConnectFuture();
+                    boolean success = connectFuture.awaitConnectionComplete(connectTimeout.getMillis(), TimeUnit.MILLISECONDS);
+                    if (success == false) {
+                        allConnected = false;
+                        Exception exception = connectFuture.getException();
+                        if (exception != null) {
+                            ex = exception;
+                            break;
+                        }
+                    }
+                }
+
+                if (allConnected == false) {
+                    if (ex == null) {
+                        throw new ConnectTransportException(node, "connect_timeout[" + connectTimeout + "]");
+                    } else {
+                        throw new ConnectTransportException(node, "connect_exception", ex);
+                    }
+                }
+                addConnectionsToList(channels, connections);
+                return true;
+            } catch (final RuntimeException e) {
```

Comment:
I changed this to be one try / catch. And it catches all RuntimeExceptions and all checked exceptions that are thrown.